### PR TITLE
This includes a few improvements described below.

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
* Added a wait for connection to the environment monitor to allow waiting on a connection to upstream environment monitor prior to proceeding. This is primarily useful for examples coming soon.
* Updated ROSPlotting to include a ros spin once and improved its wait for connection
* Moved the planner names to the request message. This is in preparation for changing the GetMotionPlan to be a vector of request to support parallelization of request. 
* The last commit restructures the planning server to improve readability and provides a good examples for setting up process managers. Before the process for setting up process managers was hard to follow.

### CODE REVIEW

File: tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h [hunk 1]
```diff
@@ -338,10 +345,10 @@ class EnvironmentMonitor
   ContinuousContactManagerPluginLoaderPtr continuous_manager_loader_;
 
   tesseract::Tesseract::Ptr tesseract_;
-  std::shared_mutex scene_update_mutex_;  /// mutex for stored scene
-  ros::Time last_update_time_;            /// Last time the state was updated
-  ros::Time last_robot_motion_time_;      /// Last time the robot has moved
-  bool enforce_next_state_update_;        /// flag to enforce immediate state update in onStateUpdate()
+  mutable std::shared_mutex scene_update_mutex_;  /// mutex for stored scene
```
Comment (mpowelson) [inline tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h:348]: Why this change?

File: tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp [hunk 2]
```diff
@@ -110,6 +110,33 @@ EnvironmentMonitor::~EnvironmentMonitor()
...
+  while (ros::ok())
+  {
+    {
+      auto lock = std::shared_lock(scene_update_mutex_);
+      is_connected = tesseract_->isInitialized();
+    }
+    if (is_connected)
+      return true;
+
+    if (wall_timeout >= ros::WallDuration(0))
```
Comment (mpowelson) [inline tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp:127]: Should this be > only to allow disabling timeout?

File: tesseract_ros/tesseract_msgs/msg/PlanningRequest.msg [hunk 3]
```diff
@@ -10,3 +10,51 @@ bool profile # Enable profiling of the planning request
 
 tesseract_msgs/PlannerProfileRemapping plan_profile_remapping
 tesseract_msgs/PlannerProfileRemapping composite_profile_remapping
+
+string TRAJOPT_PLANNER_NAME="TrajOptPlanner"
+string OMPL_PLANNER_NAME="OMPLPlanner"
+string DESCARTES_PLANNER_NAME="DescartesPlanner"
+string FREESPACE_PLANNER_NAME = "FreespacePlanner"
+string CARTESIAN_PLANNER_NAME = "CartesianPlanner"
```
Comment (mpowelson) [inline tesseract_ros/tesseract_msgs/msg/PlanningRequest.msg:18]: Formatting

File: tesseract_ros/tesseract_rosutils/src/plotting.cpp [hunk 4]
```diff
@@ -64,23 +64,35 @@ bool ROSPlotting::isConnected() const { return true; }
...
+  while (ros::ok())
   {
-    if (!isConnected())
-      sleep(1);
-    else
-      break;
+    if (isConnected())
+      return;
+
+    if (wall_timeout >= ros::WallDuration(0))
```
Comment (mpowelson) [inline tesseract_ros/tesseract_rosutils/src/plotting.cpp:74]: Same here. >?

**Review by mpowelson — COMMENTED** [review]
Looks reasonable.

File: tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/include/tesseract_monitoring/environment_monitor.h:348]: Need to change it from a const function, because I need to lock it for read.

File: tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp (hunk 2 above)
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp:127]: I took a look at the waitForService code and this is what they did. The default value is `ros::Duration(-1)`, but I assume someone could pass in `ros::Duration(0)` which would indicate only to run the check once and then return. If it was greater than only it would wait indefinitely.

File: tesseract_ros/tesseract_msgs/msg/PlanningRequest.msg (hunk 3 above)
Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_msgs/msg/PlanningRequest.msg:18]: fixing.

**Review by mpowelson — APPROVED** [review]
